src/Plugins/ComplexCore/src/ComplexCore/Filters/ReplaceElementAttributesWithNeighborValuesFilter.cpp

```diff
@@ -0,0 +1,106 @@
+}
+
+//------------------------------------------------------------------------------
+std::string ReplaceElementAttributesWithNeighborValuesFilter::className() const
+{
+  return FilterTraits<ReplaceElementAttributesWithNeighborValuesFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid ReplaceElementAttributesWithNeighborValuesFilter::uuid() const
+{
+  return FilterTraits<ReplaceElementAttributesWithNeighborValuesFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string ReplaceElementAttributesWithNeighborValuesFilter::humanName() const
+{
+  return "Replace Element Attributes with Neighbor (Threshold)";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> ReplaceElementAttributesWithNeighborValuesFilter::defaultTags() const
+{
+  return {"#Processing", "#Cleanup"};
+}
+
+//------------------------------------------------------------------------------
+Parameters ReplaceElementAttributesWithNeighborValuesFilter::parameters() const
+{
+  Parameters params;
+
+  params.insertSeparator(Parameters::Separator{"Input Parameters"});
+
+  params.insert(std::make_unique<Float32Parameter>(k_MinConfidence_Key, "Threshold Value", "", 0.1F));
+  params.insert(std::make_unique<ChoicesParameter>(k_SelectedComparison_Key, "Comparison Operator", "", 0, ChoicesParameter::Choices{"< [Less Than]", "> [Greater Than]"}));
+  params.insert(std::make_unique<BoolParameter>(k_Loop_Key, "Loop Until Gone", "", false));
+
+  params.insertSeparator(Parameters::Separator{"Required Input Cell Data"});
+  params.insert(std::make_unique<GeometrySelectionParameter>(k_SelectedImageGeometry_Key, "Selected Image Geometry", "The target geometry", DataPath{},
+                                                             GeometrySelectionParameter::AllowedTypes{IGeometry::Type::Image}));
+  params.insert(
+      std::make_unique<ArraySelectionParameter>(k_ConfidenceIndexArrayPath_Key, "Comparison Array", "", DataPath{}, complex::GetAllDataTypes(), ArraySelectionParameter::AllowedComponentShapes{{1}}));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer ReplaceElementAttributesWithNeighborValuesFilter::clone() const
+{
+  return std::make_unique<ReplaceElementAttributesWithNeighborValuesFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult ReplaceElementAttributesWithNeighborValuesFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                                                         const std::atomic_bool& shouldCancel) const
+{
+  auto pMinConfidenceValue = filterArgs.value<float32>(k_MinConfidence_Key);
```

Comment:
```suggestion
  auto pMinConfidenceValue = filterArgs.value<float32>(k_MinConfidence_Key);
```
These copies are unused and can probably be removed